src/test/java/seedu/address/model/InterestTest.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model;
+
+import static org.junit.Assert.assertEquals;
+import static seedu.address.model.util.DateUtil.generateOutdatedDebtDate;
+
+import java.util.Date;
+
+import org.junit.Test;
+
+import seedu.address.model.person.Person;
+import seedu.address.model.util.DateUtil;
+import seedu.address.testutil.PersonBuilder;
+
+public class InterestTest {
+    // dates that interest should affect debt
+    private final String sampleDateInput1 = "01-11-2017";
+    private final String sampleDateInput2 = "01-01-2017";
+    private final String sampleDateInput3 = "01-03-2017";
+    private final String sampleDateInput4 = "01-05-2017";
+    private final String sampleDateInput5 = "01-04-2017";
+
+    // dates that interest should not affect debt
+    private final String sampleDateInput6 = "20-05-2017";
+
+    @Test
+    public void checkUpdateDebtTest() {
+        // personToTest has interest rate of 1% and debt of $10,000
+        Person personToTest = new PersonBuilder().withDebt("10000").build();
+        Date lastAccruedDate = generateOutdatedDebtDate(generateDateFromString(sampleDateInput1));
+        personToTest.setLastAccruedDate(lastAccruedDate);
+        assertEquals(personToTest.checkUpdateDebt(generateDateFromString(sampleDateInput1)), 1);
+        lastAccruedDate = generateOutdatedDebtDate(generateDateFromString(sampleDateInput2));
```

Comment:
perhaps a newline can be inserted after each block of test for each sample data input